atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -199,24 +213,29 @@ private long getBackoffTimeWhenSweepHasNotRun() {
 
     @VisibleForTesting
     SweepOutcome runOnce() {
-        Optional<TableToSweep> tableToSweep = getTableToSweep();
-        if (!tableToSweep.isPresent()) {
+        Optional<TableToSweep> tableToSweepOptional = getTableToSweep();
+        if (!tableToSweepOptional.isPresent()) {
             // Don't change this log statement. It's parsed by test automation code.
             log.debug("Skipping sweep because no table has enough new writes to be worth sweeping at the moment.");
             return SweepOutcome.NOTHING_TO_SWEEP;
         }
 
+        TableToSweep tableToSweep = tableToSweepOptional.get();
         SweepBatchConfig batchConfig = sweepBatchConfigSource.getAdjustedSweepConfig();
+        if (StreamTableType.isStreamStoreValueTable(tableToSweep.getTableRef())) {
```

Comment:
I think this is a bit odd.  We override the config here, but we're still decreasing the basic sweep config behind the scenes, eventually we get to the point where this has decreased below the onfig here and we defer to using that until it increases again.

This has a few problems.  Metrics on the batchSizeMultiplier will be wrong at times, logging when we decrease the metrics is incorrect and we won't backoff on failure for the first 4 failures on the stream store.

If we push the override config down into the `AdjustableBatchConfigSource` (just rename it `SweepBatchConfigSource`) then we hide the complexity in there, we can get metrics and logging right and backoffs work as normal.  We can pass in the table name and it can figure out what table type it is.

This has some nice properties in the future too - we can have other special cases for other table types and we can even remember and reset the batch config when we change table.

Reply:
Yeah, agreed. I would expect the batch size to look like 128, 64, 32... for streams and 1024, 512, 256... for normal tables.